### DISCUSSION TITLE
Where to add SqlServer connection string to existing Azure Sql Server DB?

### DISCUSSION THREAD

**Discussion by niclascoeli** [microsoft/aspire#3817] [Q&A]
Not sure how to set up the proper configuration in Azure? Previously I just used env variable and got the config directly but after moving to Aspire Preview 5 i thought i should do it the proper way.

My code (works well in debug):
![image](https://github.com/Azure/azure-dev/assets/141612759/3c00da7d-f527-4a97-9c4c-d34c75417d74)

The exception in Azure, I already have the DB set up and would like to know where I should set the connection string. I guess the message is intended to be helpful but I don't understand how to translate this to configuration in Container Apps. I tried setting env variable AZURE_CONNECTION_STRING but no cigar..

```c#
An unhandled exception has occurred while executing the request.
2024-04-18T14:15:48.391718948Z       System.InvalidOperationException: ConnectionString is missing. It should be provided in 'ConnectionStrings:<REDACTED>' or under the 'ConnectionString' key in 'Aspire:Microsoft:EntityFrameworkCore:SqlServer' or 'Aspire:Microsoft:EntityFrameworkCore:SqlServer:<REDACTED>DbContext' configuration section.
```

**Comment by davidfowl** [comment]
Can you show the app host code?

**Reply by niclascoeli** [reply]
Sure! I thought it wouldn't matter as it seemed to hint towards a generic environment variable. Here it is:

```c#
var builder = DistributedApplication.CreateBuilder(args);

var authId = builder.Configuration["AUTH_MS_CLIENTID"];
var authSecret = builder.Configuration["AUTH_MS_CLIENTSECRET"];

var sql = builder.AddSqlServer("<REDACTED_S1>")
                 .PublishAsAzureSqlDatabase()
                 .AddDatabase(<REDACTED_S2>);

#if DEBUG
builder.AddProject<Projects.<REDACTED>_MigrationService>("migration").WithReference(sql);
#endif

var web = builder.AddProject<Projects.<REDACTED>>("<REDACTED_S3>")
                      .WithReference(sql)
                      .WithEnvironment("AUTH_MS_CLIENTID", authId)
                      .WithEnvironment("AUTH_MS_CLIENTSECRET", authSecret);

builder.Build().Run();
```

**Reply by timheuer** [reply]
Does looking at this sample help: https://github.com/dotnet/aspire-samples/blob/main/samples/DatabaseMigrations/DatabaseMigrations.ApiService/Program.cs#L11 (you can look at the apphost to see how it is configured like yours...)
